token-syncer/src/token_syncer/commands/ping.clj

```diff
@@ -0,0 +1,98 @@
+;;
+;; Copyright (c) Two Sigma Open Source, LLC
+;;
+;; Licensed under the Apache License, Version 2.0 (the "License");
+;; you may not use this file except in compliance with the License.
+;; You may obtain a copy of the License at
+;;
+;;  http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+;;
+(ns token-syncer.commands.ping
+  (:require [clojure.pprint :as pp]
+            [clojure.string :as str]
+            [clojure.tools.logging :as log]
+            [plumbing.core :as pc]))
+
+(defn ping-token
+  "Pings the specified tokens across provided clusters based on cluster-urls and returns the result.
+   Throws an exception if there was an error while pinging the clusters with the provided token."
+  [{:keys [health-check-token]} cluster-urls token]
+  (try
+    (log/info "pinging token" token "on clusters:" cluster-urls)
+    (let [cluster-url->result
+          (pc/map-from-keys
+            (fn [cluster-url]
+              (try
+                (let [{:keys [body headers status]} (health-check-token cluster-url token)]
+                  (log/info "health check repsonse:"
+                            {:body (str/trim (str body))
+                             :headers headers
+                             :status status})
+                  (if (and (integer? status) (<= 200 status 299))
+                    {:exit-code 0
+                     :message (str "Successfully pinged token " token " on " cluster-url
+                                   ", reason: health check returned status code " status)}
+                    {:exit-code 1
+                     :message (str "unable to ping token " token " on " cluster-url
+                                   ", reason: health check returned status code " status)}))
```

Comment:
Should we not return the token, cluster URL, and status as fields in the return map?

Reply:
I chose to log status and body instead of returning it as it is not used upstream in the caller.

The sub-command operators on a single token at a time, so I did not include the token in the map.

Reply:
Got it